transaction/core/src/tx_error.rs

```diff
@@ -84,6 +84,8 @@ pub enum NewMemoError {
     Creation(MemoError),
     /// Utf-8 did not properly decode
     Utf8Decoding,
+    /// Max fee of {0} exceeded. Attempted to set fee amount: {1}
+    MaxFeeExceeded(u64, u64),
```

Comment:
In the other memo builder we were using the `LimitsExceeded` error with string `"fee"`, but we could change those to use this error instead?

Reply:
I i'd propose maybe something like 

```rust
LimitsExceeded('static &str, u64, u64)
```

i.e. domain/max val/attempted val - thoughts?

Reply:
sounds good to me!

Reply:
Actually would you mind if I did this in a separate PR? It involves changing some of the plumbing of the other memo builders and I'd like to catch any breakage in that scope.